library/core/src/iter/adapters/intersperse.rs

```diff
@@ -0,0 +1,76 @@
+use super::Peekable;
+
+/// An iterator adapter that places a separator between all elements.
+#[unstable(feature = "iter_intersperse", reason = "recently added", issue = "none")]
+#[derive(Debug, Clone)]
+pub struct Intersperse<I: Iterator>
+where
+    I::Item: Clone,
+{
+    separator: I::Item,
+    iter: Peekable<I>,
```

Comment:
I believe `iter` should comes first since it is the most important thing?
```suggestion
    iter: Peekable<I>,
    separator: I::Item,
```